### PR TITLE
Add options for scripting with daml start

### PR DESCRIPTION
**Pull request by associahedron**
Added three options to `daml start`:

* `--start-navigator` to control whether navigator is started (defaults to yes)
* `--on-start` to run a command after sandbox and navigator are started (before the browser is opened, which is optional anyway)
* `--wait-for-signal` to control whether to wait for Ctrl+C or an interrupt at the end (defaults to yes)

TODO (later once sandbox ports / navigator ports get determined automatically): pass the ports in the environment

### CODE REVIEW

File: daml-assistant/daml-helper/src/DamlHelper.hs [hunk 1]
```diff
@@ -532,8 +534,14 @@ withNavigator (SandboxPort sandboxPort) navigatorPort config args a = do
 -- | Whether `daml start` should open a browser automatically.
 newtype OpenBrowser = OpenBrowser Bool
 
-runStart :: OpenBrowser -> IO ()
-runStart (OpenBrowser shouldOpenBrowser) = withProjectRoot Nothing (ProjectCheck "daml start" True) $ \_ _ -> do
+-- | Whether `daml start` should start the navigator automatically.
+newtype StartNavigator = StartNavigator Bool
+
+-- | Whether `daml start` should wait for Ctrl+C or interrupt after starting servers.
+newtype WaitForSignal = WaitForSignal Bool
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DamlHelper.hs]: What is the usecase for this?

Comment (associahedron) [reply daml-assistant/daml-helper/src/DamlHelper.hs]: For when you want to run a script using `--on-start` and then quit after the script finishes (i.e. for testing).

Comment (cocreature) [reply daml-assistant/daml-helper/src/DamlHelper.hs]: Thanks, sounds like a reasonable usecase. An alternative worth considering might be to exit if the start script fails with a non-zero exit code but I don’t feel strongly about this so your choice.

Comment (associahedron) [reply daml-assistant/daml-helper/src/DamlHelper.hs]: Seems reasonable to me to make `daml start` exit with the exit code of the `--on-start` command whenever it's non-zero.

**Pull request merged**